EazyE2E/Process/EzProcess.cs

```diff
@@ -1,42 +1,69 @@
 ﻿using System;
 using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
 
 namespace EazyE2E.Process
 {
-    public class EzProcess
+    public class EzProcess : IDisposable
     {
+        private readonly string _processFullPath;
         private readonly string _processName;
-        private readonly string _arguments;
-        private readonly ProcessWindowStyle _winStyle;
+        private System.Diagnostics.Process _process;
 
-        public EzProcess(string processName, string arguments = "", ProcessWindowStyle winStyle = ProcessWindowStyle.Normal)
```

Comment:
Optional parameters don't really belong in a constructor.  The are better suited for public properties.  This gives the user the ability to still configure the object while only requiring what is necessary to instantiate the object in the constructor.